web/backend/src/Server.ts

```diff
@@ -216,10 +216,23 @@ app.post('/api/add_role', (req, res) => {
   const { sciper } = req.body;
   const { role } = req.body;
 
-  usersDB.put(sciper, role).catch((error) => {
-    res.status(500).send('Failed to add role');
-    console.log(error);
-  });
+  // The sciper has to contain 6 numbers
+  if (sciper < 999999 && sciper > 100000) {
+    // call https://search-api.epfl.ch/api/ldap?q=228271 if the answer
+    // empty then sciper invalid
+    axios.get(`https://search-api.epfl.ch/api/ldap?q=${sciper}`).then((response) => {
+      if (response.data.length === 0) {
+        res.status(400).send('Unknown sciper');
+      } else {
+        usersDB.put(sciper, role).catch((error) => {
+          res.status(500).send('Failed to add role');
+          console.log(error);
+        });
+      }
+    });
+  } else {
+    res.status(400).send('sciper length incorrect');
+  }
```

Comment:
In general, you can reduce indentation by using guard clauses and making the code more readable.
Just a suggestion:

```suggestion
  // The sciper has to contain 6 numbers
  if (sciper > 999999 || sciper < 100000) {
    res.status(400).send('sciper length incorrect');
    return;
  }
  
  // call https://search-api.epfl.ch/api/ldap?q=228271 
  // if the answer is empty then the sciper is invalid
  axios.get(`https://search-api.epfl.ch/api/ldap?q=${sciper}`).then((response) => {
    if (response.data.length === 0) {
      res.status(400).send('Unknown sciper');
    } else {
      usersDB.put(sciper, role).catch((error) => {
        res.status(500).send('Failed to add role');
        console.log(error);
      });
    }
  });
```